### PR TITLE
Add active page styling and hover styling to navigation links

### PR DESCRIPTION
**Pull request by ccoutts2**
Simple active page and hover state styling for nav links. 

We can adjust when he refactor styling

Edited nav link component to only take in label props as link label (before was href || label).
Added label to each of the navlinks in the navbar component.

I had to do this to get usePathname() hook to work

### CODE REVIEW

File: src/components/navigation/NavLink/NavLink.tsx [hunk 1]
```diff
@@ -1,19 +1,27 @@
 import Link from "next/link";
 
+import { usePathname } from "next/navigation";
+
 interface NavLinkProps {
   href: string;
   label?: string;
```
Comment (kabirseth4) [inline src/components/navigation/NavLink/NavLink.tsx]: Need to make this non-optional

File: src/components/navigation/NavLink/NavLink.tsx [hunk 2]
```diff
@@ -1,19 +1,27 @@
...
+
   return (
     <Link
       className="inline-block w-full py-2 text-sm uppercase lg:w-fit lg:py-3 lg:text-lg"
       href={href}
       onMouseEnter={onMouseEnter}
     >
-      {href || label}
+      <span
+        className={`hover:border-b-2 hover:border-[#5d3a40] ${pathname === href ? "border-b-2 border-[#5d3a40]" : ""}`}
```
Comment (kabirseth4) [inline src/components/navigation/NavLink/NavLink.tsx]: If I go to /schemes/bedroom, schemes is not underlined. Should it be if pathname.startsWith(href)?

**Review by kabirseth4 — CHANGES_REQUESTED** [review]
Two smol changes

**Review by kabirseth4 — CHANGES_REQUESTED** [review]
Two smol changes

**Comment by ccoutts2** [comment]
@kabirseth4 

Changes

- Remove optional prop tag from label
- Refactor pathname so dynamic pages from schemes page attach the pathname styling to the schemes nav link

**Review by kabirseth4 — APPROVED** [review]
All looks good to me. Will merge

**Pull request merged**